server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -1931,62 +1948,237 @@ public void deletePool(Pool pool) {
+                    // Since we're sifting through these already, let's also sort them into consumer lists
+                    // for some of the other methods we'll be calling later
+                    Consumer consumer = entitlement.getConsumer();
+                    Pool pool = entitlement.getPool();
+
+                    List<Entitlement> stackedEntitlements = consumerEntitlements.get(consumer);
+                    if (stackedEntitlements == null) {
+                        stackedEntitlements = new LinkedList<Entitlement>();
+                        consumerEntitlements.put(consumer, stackedEntitlements);
+                    }
 
-        // Look up the related pools for these subscription IDs.
-        if (!subscriptionIds.isEmpty()) {
-            poolsToDelete.addAll(this.poolCurator.getPoolsBySubscriptionIds(subscriptionIds).list());
-        }
+                    if (!"true".equals(pool.getAttributeValue(Pool.Attributes.DERIVED_POOL)) &&
+                        pool.hasProductAttribute(Product.Attributes.STACKING_ID)) {
 
-        if (!poolsToDelete.isEmpty()) {
-            revokeEntitlements(entitlementsToRevoke, alreadyDeletedPools);
-            log.debug("Batch deleting pools after successful revocation");
-            poolCurator.batchDelete(poolsToDelete, alreadyDeletedPools);
-        }
+                        stackedEntitlements.add(entitlement);
+                    }
 
-        for (Pool pool : poolsToDelete) {
-            Event event = eventFactory.poolDeleted(pool);
-            sink.queueEvent(event);
+                    // Update quantities if the entitlement quantity is non-zero
+                    int quantity = entitlement.getQuantity() != null ? entitlement.getQuantity() : 0;
+                    if (quantity != 0) {
+                        // Update the pool quantities if we didn't delete it
+                        if (affectedPoolIds.contains(pool.getId())) {
+                            pool.setConsumed(pool.getConsumed() - quantity);
+                            poolsToSave.add(pool);
+                        }
+
+                        // Update entitlement counts for affected consumers...
+                        consumer.setEntitlementCount(consumer.getEntitlementCount() - quantity);
+                        if (consumer.getType().isManifest()) {
+                            pool.setExported(pool.getExported() - quantity);
+                        }
+                    }
+                }
+
+                this.poolCurator.updateAll(poolsToSave, false, false);
+                this.consumerCurator.updateAll(consumerEntitlements.keySet(), false, false);
+                this.consumerCurator.flush();
+                log.info("Entitlement counts successfully updated for {} pools and {} consumers",
+                    poolsToSave.size(), consumerEntitlements.size());
+
+                // Update stacked entitlements for affected consumers(???)
```

Comment:
The ??? makes me sense hesitation or question on your part. Any reason why?

Reply:
I don't recall at this point. I think I was confused as to why that's needed in this case, what it's actually doing or whether or not it's an efficient way of completing that task.